### ISSUE TITLE
plot(xts.object, col = ifelse()) not working in 0.14.1

### ISSUE THREAD

**Issue by dnousias** [joshuaulrich/xts#426]
### Description

Upgrading to xts 0.14.1 doesn't allow xts plot where the col argument has a value for every row in the xts object. This works in xts version 0.14.0.

### Minimal, reproducible example

```r
n <- 1000
series <- rnorm(n)
datetimes <- seq(as.POSIXct("2017-03-27"), length.out = n, by = "days")

library(xts)
x <- xts(series, datetimes)
colors <- ifelse(x >=0 , "green", "red")
plot(x, col = colors, type = "h")
```

First, in xts version 0.14.1:

![image](https://github.com/user-attachments/assets/7d13ae21-5137-4adf-9a91-23c7fc325075)

But in xts version 0.14.0:

![image](https://github.com/user-attachments/assets/8487a658-43c4-464f-ae0f-666d5e557fa6)

**Comment by ethanbsmith** [comment]
did i break this w/ #423? if so, u can assign to me

**Comment by joshuaulrich** [comment]
@ethanbsmith looks like it. It gives the expected result in the commit prior to the fix (6523b545926efa2cc14cca5b6a4fc4ec00cb0763) and is broken in the commit with the fix (3086e140f126b67fd0fc877595c4f0730f7fe1f1).
